src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -2364,9 +2364,11 @@ void MethodContext::recGetHelperFtn(CorInfoHelpFunc ftnNum, void** ppIndirection
     {
         DLDL oldValue = GetHelperFtn->Get(key);
 
-        AssertCodeMsg(oldValue.A == value.A && oldValue.B == oldValue.B, EXCEPTIONCODE_MC,
-                      "collision! old: %016" PRIX64 " %016" PRIX64 ", new: %016" PRIX64 " %016" PRIX64 " \n", oldValue.A, oldValue.B, value.A,
-                      value.B);
+        // We can't use string concatenation in an argument to the `AssertCodeMsg` macro, so construct the string here.
+        char assertMsgBuff[200];
+        sprintf_s(assertMsgBuff, sizeof(assertMsgBuff), "old: %016" PRIX64 " %016" PRIX64 ", new: %016" PRIX64 " %016" PRIX64,
+            oldValue.A, oldValue.B, value.A, value.B);
+        AssertCodeMsg(oldValue.A == value.A && oldValue.B == oldValue.B, EXCEPTIONCODE_MC, "collision! %s", assertMsgBuff);
```

Comment:
I think we should just fix `AsSertCodeMsg` like this:
```diff
- LogException(exCode, "SuperPMI assertion '%s' failed (" #msg ")", #expr, ##__VA_ARGS__);
+ LogException(exCode, "SuperPMI assertion '%s' failed (" msg ")", #expr, ##__VA_ARGS__);
```

Currently it is stringizing the format string, which is almost certainly not correct.

Reply:
Makes sense. Updated.